romancal/assign_wcs/utils.py

```diff
@@ -358,29 +358,30 @@ def compute_fiducial(wcslist, bounding_box=None):
     return fiducial
 
 
-def add_s_region(model):
-    """
-    Calculate the detector's footprint using ``WCS.footprint`` and save it in the
-    ``S_REGION`` keyword
+def create_footprint(wcs, shape=None):
+    """Calculate sky footprint
 
     Parameters
     ----------
-    model : `~roman_datamodels.datamodels.ImageModel`
-        The data model for processing
+    wcs : `gwcs.WCS`
+        The WCS information to get the footprint from
+
+    shape : n-tuple or None
+       Shape to use if wcs has no defined shape.
 
     Returns
     -------
-    A formatted string representing the detector's footprint
+    footprint : `numpy.ndarray`
+        The footprint.
     """
-
-    bbox = model.meta.wcs.bounding_box
+    bbox = wcs.bounding_box
 
     if bbox is None:
-        bbox = wcs_bbox_from_shape(model.data.shape)
+        bbox = wcs_bbox_from_shape(shape)
 
     # footprint is an array of shape (2, 4) - i.e. 4 values for RA and 4 values for
     # Dec - as we are interested only in the footprint on the sky
-    footprint = model.meta.wcs.footprint(bbox, center=True, axis_type="spatial").T
+    footprint = wcs.footprint(bbox, center=True, axis_type="spatial").T
```

Comment:
Just out of curiosity, should we always use the center of the pixel (`center=True`) when creating the footprint? Wouldn't it be better to allow that parameter to be passed in when calling `create_footprint()`, and set its default value to `True`?